idaes/models_extra/power_generation/unit_models/heater_1D.py

```diff
@@ -0,0 +1,564 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+1-D Electric Trim Heater Model With Wall Temperatures
+
+Discretization based on tube rows
+"""
+# Import Pyomo libraries
+from pyomo.environ import (
+    assert_optimal_termination,
+    Var,
+    value,
+    units as pyunits,
+)
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+# Import IDAES cores
+from idaes.core import (
+    ControlVolume1DBlock,
+    declare_process_block_class,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+    FlowDirection,
+    UnitModelBlockData,
+    useDefault,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.solvers import get_solver
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.misc import add_object_reference
+import idaes.logger as idaeslog
+from idaes.core.util.tables import create_stream_table_dataframe
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.models_extra.power_generation.unit_models.heat_exchanger_common import (
+    _make_geometry_common,  # pylint: disable=W0212
+    _make_performance_common,  # pylint: disable=W0212
+    _scale_common,  # pylint: disable=W0212
+)
+
+__author__ = "Jinliang Ma, Douglas Allan"
+
+
+@declare_process_block_class("Heater1D")
+class Heater1DData(UnitModelBlockData):
+    """Standard Trim Heater Model Class Class."""
+
+    CONFIG = ConfigBlock()
```

Comment:
I would recommend using the `CONFIG` from `UnitModelBlockData` as it already has the `dynamic` and `has_holdup` options, and means that if new required arguments are added in the future that these will be inherited automatically.